src/components/textarea/TextArea.spec.tsx

```diff
@@ -0,0 +1,187 @@
+import {HTMLAttributes, mount, ReactWrapper, shallow} from 'enzyme';
+import * as React from 'react';
+import {Provider} from 'react-redux';
+import {Store} from 'react-redux';
+import {findWhere} from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {clearState} from '../../utils/ReduxUtils';
+import {TestUtils} from '../../utils/TestUtils';
+import {ITextAreaProps, TextArea, TextAreaConnected} from './TextArea';
+import {ITextAreaState} from './TextAreaReducers';
+
+describe('TextArea', () => {
+    describe('<TextArea />', () => {
+        it('should render without errors', () => {
+            expect(() => {
+                shallow(
+                    <TextArea id='textarea-id' />,
+                );
+            }).not.toThrow();
+        });
+    });
+
+    describe('<TextArea />', () => {
+        let wrapper: ReactWrapper<ITextAreaProps, any>;
+        let textArea: ReactWrapper<HTMLAttributes, any>;
+
+        beforeEach(() => {
+            wrapper = mount(
+                <TextArea id='textarea-id' />,
+                {attachTo: document.getElementById('App')},
+            );
+
+            textArea = wrapper.find('textarea').first();
+        });
+
+        afterEach(() => {
+            wrapper.unmount();
+            wrapper.detach();
+        });
+
+        it('should set textarea id when specified', () => {
+            expect(textArea.prop('id')).toBe('textarea-id');
+        });
+
+        it('should set className when specified', () => {
+            const className = 'a-class';
+            expect(textArea.hasClass(className)).toBe(false);
+
+            wrapper.setProps({className}).mount();

```

Comment:
Not sure of what I'm saying here, but I think the call to mount after setProps is unecessary (here and other places in the file)

Reply:
removed where useless